ruby/tests/repeated_field_test.rb

```diff
@@ -0,0 +1,577 @@
+    m = TestMessage.new
+    5.times{|i| m.repeated_string << 'string' }
+    count = 0
+    m.repeated_string.each do |val|
+      assert_equal 'string', val
+      count += 1
+    end
+    assert_equal 5, count
+    result = m.repeated_string.each{|val| val + '_junk'}
+    assert_equal ['string'] * 5, result
+  end
+
+
+  def test_empty?
+    m = TestMessage.new
+    assert_equal true, m.repeated_string.empty?
+    m.repeated_string << 'foo'
+    assert_equal false, m.repeated_string.empty?
+    m.repeated_string << 'bar'
+    assert_equal false, m.repeated_string.empty?
+  end
+
+  def test_array_accessor
+    m = TestMessage.new
+    reference_arr = %w(foo bar baz)
+    m.repeated_string += reference_arr.clone
+
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr[1]
+    end
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr[-2]
+    end
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr[20]
+    end
+    # TODO: accessor doesn't allow other ruby-like methods
+    # test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+    #   arr[1, 2]
+    # end
+    # test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+    #   arr[0..2]
+    # end
+    # test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+    #   arr[-1, 1]
+    # end
+    # test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+    #   arr[10, 12]
+    # end
+  end
+
+  def test_array_settor
+    m = TestMessage.new
+    reference_arr = %w(foo bar baz)
+    m.repeated_string += reference_arr.clone
+
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr[1] = 'junk'
+    end
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr[-2] = 'snappy'
+    end
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr[3] = ''
+    end
+    # slight deviation; we are strongly typed, and nil is not allowed
+    # for string types;
+    m.repeated_string[5] = 'spacious'
+    assert_equal ["foo", "snappy", "baz", "", "", "spacious"], m.repeated_string
+    # test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+    #   arr[20] = 'spacious'
+    # end
+    # TODO: accessor doesn't allow other ruby-like methods
+    # test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+    #   arr[1, 2] = 'fizz'
+    # end
+    # test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+    #   arr[0..2] = 'buzz'
+    # end
+  end
+
+  def test_push
+    m = TestMessage.new
+    reference_arr = %w(foo bar baz)
+    m.repeated_string += reference_arr.clone
+
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr.push('fizz')
+    end
+    test_self_modifying_method(m.repeated_string, reference_arr) do |arr|
+      arr << 'fizz'
+    end
+    #TODO: push should support multiple
```

Comment:
I documented the differences between `RepeatedField` behavior vs `Array` behavior in comments like this